feat(mep): Split the metrics compatibility endpoint in two

wmak opened the pull request:
- This moves the sums and compatible project checks in two so the
  frontend can either run them in parallel if needed or conditionally
  run one based on the other
- This deprecates the old endpoint, it will be deleted once the frontend
  no longer uses it

wmak committed:
feat(mep): Split the metrics compatibility endpoint in two

- This moves the sums and compatible project checks in two so the
  frontend can either run them in parallel if needed or conditionally
  run one based on the other
- This deprecates the old endpoint, it will be deleted once the frontend
  no longer uses it

k-fish commented on src/sentry/api/endpoints/organization_metrics_meta.py:
```diff
@@ -0,0 +1,113 @@
+                functions_acl=["count_null_transactions", "count_has_transaction_name"],
+                use_aggregate_conditions=True,
+            )
+            data["compatible_projects"] = sorted(
+                row["project.id"] for row in compatible_results["data"]
+            )
```
Can you add `incompatible_projects` so we can check the type of incompatibility? Or do we want to wait until the sdk's are updated?

wmak committed:
ref: Add an incompatible projects property

- Follows the per_page arameter but intentionally not implementing
  pagination to keep this simple for now
